test/python/algorithms/quantum_time_evolution/variational/calculators/test_metric_tensor_calculator.py

```diff
@@ -0,0 +1,786 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test metric tensor calculator."""
+
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+import numpy as np
+from numpy import array
+
+from qiskit.algorithms.quantum_time_evolution.variational.calculators.metric_tensor_calculator import (
+    calculate,
+)
+from qiskit.circuit.library import EfficientSU2
+from qiskit.opflow import SummedOp, X, Y, I, Z
+
+
+class TestMetricTensorCalculator(QiskitAlgorithmsTestCase):
+    """Test metric tensor calculator."""
+
+    # checked, correct
+    def test_calculate_real(self):
+        """Test calculating the real part of a metric tensor."""
+        observable = SummedOp(
+            [
+                0.2252 * (I ^ I),
+                0.5716 * (Z ^ Z),
+                0.3435 * (I ^ Z),
+                -0.4347 * (Z ^ I),
+                0.091 * (Y ^ Y),
+                0.091 * (X ^ X),
+            ]
+        ).reduce()
+
+        d = 2
+        ansatz = EfficientSU2(observable.num_qubits, reps=d)
+
+        # Define a set of initial parameters
+        parameters = ansatz.ordered_parameters
+        metric_tensor = calculate(ansatz, parameters)
+
+        values_dict = [
+            {param: np.pi / 4 for param in parameters},
+            {param: np.pi / 2 for param in parameters},
+        ]
+        # TODO extract
+        correct_values = [
+            array(
+                [
+                    [
+                        1.00000000e00 + 0.0j,
```

Comment:
If I remember correctly, these values come from the prototype. So we made sure these results match those from the prototype. It wouldn't hurt to also verify it by hand, though. Are you volunteering? :)

Reply:
I did some once, I can look for my notes 😉 Otherwise we can also use https://github.com/Cryoris/surfer as reference, which does a classical calculation of the QFI